lib/round.rb

```diff
@@ -0,0 +1,145 @@
+class Round
+
+  attr_accessor :deck, :turns
+
+  def initialize(deck)
+    @deck = deck
+    @turns = []
+  end
+
+  def current_card
+    deck.cards.first
+  end
+
+  def take_turn(guess)
+    turn = Turn.new(guess, deck.cards.first)
+    turns << turn
+    deck.cards.rotate!
+    return turn
+  end
+
+  def correct?
+    round.turns.first.guess == round.turns.fist.card.answer
+  end
+
+  def number_correct
+    num_correct = 0
+    turn_index = 0
+    until turn_index == @turns.count
+      turns.each do |turn|
+        if turns[turn_index].guess == turns[turn_index].card.answer
+          num_correct += 1
+          turn_index += 1
+        else
+          turn_index += 1
+        end
+      end
+    end
+    num_correct
+  end
+
+  def number_correct_by_category(category)
+    num_correct = 0
+    turn_index = 0
+    until turn_index == @turns.count
+      @turns.each do |turn|
+        if @turns[turn_index].card.category == category && @turns[turn_index].guess == @turns[turn_index].card.answer
```

Comment:
Similar to the above comment; comparing direct matches with guesses & answers provides a worse user experience than if we compare against `guess.downcase`, for example.